lib/multistream.js

```diff
@@ -77,22 +78,41 @@ function multistream (streamsArray, opts) {
   }
 
   function add (dest) {
+    if (!dest) {
+      return res
+    }
+
+    // Check that dest implements either StreamEntry or DestinationStream
+    const isStream = typeof dest.write === 'function' || dest.stream
+    const stream_ = dest.write ? dest : dest.stream
+    // This is necessary to provide a meaningful error message, otherwise it throws somewhere inside write()
+    if (!isStream) {
+      throw Error('stream object needs to implement either StreamEntry or DestinationStream interface')
+    }
+
     const { streams } = this
-    if (typeof dest.write === 'function') {
-      return add.call(this, { stream: dest })
-    } else if (typeof dest.levelVal === 'number') {
-      return add.call(this, Object.assign({}, dest, { level: dest.levelVal, levelVal: undefined }))
-    } else if (typeof dest.level === 'string') {
-      return add.call(this, Object.assign({}, dest, { level: levels[dest.level] }))
-    } else if (typeof dest.level !== 'number') {
-      // we default level to 'info'
-      dest = Object.assign({}, dest, { level: 30 })
-    } else {
-      dest = Object.assign({}, dest)
+    const level =
+      // levelVal overrides level property, if provided
+      typeof dest.levelVal === 'number'
+        ? dest.levelVal
+        // Otherwise, if level property is provided as string
+        : typeof dest.level === 'string'
+          ? levels[dest.level]
+          // Otherwise, use it as is if a number
+          : typeof dest.level === 'number'
+            ? dest.level
+            // Default fallback level
+            : DEFAULT_INFO_LEVEL
```

Comment:
Would you mind to restore the previous if/else block?

Reply:
of course, I restored the if/else block instead of the conditional operator - unless you meant reverting the whole change?